# Correlation

Relationship between an independent and dependent variable.

Types of correlation
1. Positive: The dependent and independent variable are changing in the same direction (if one increases another increases too and vice versa) 
2. Negative: The dependent and independent variable are changing in the oppostive direction (if one increases another decreases and vice versa) 
3. Zero: The dependent variable is unaffected by the change in independent variable

Checking correlation between two variables 

In [1]:
# import necessary libararies 
import seaborn as sb 
df = sb.load_dataset('tips') 
df.head()

total_bill   tip     sex smoker  day    time  size
0       16.99  1.01  Female     No  Sun  Dinner     2
1       10.34  1.66    Male     No  Sun  Dinner     3
2       21.01  3.50    Male     No  Sun  Dinner     3
3       23.68  3.31    Male     No  Sun  Dinner     2
4       24.59  3.61  Female     No  Sun  Dinner     4

In [2]:
# Checking correlation between tip and total bill 
df[['tip', 'total_bill']].corr()

tip  total_bill
tip         1.000000    0.675734
total_bill  0.675734    1.000000

<b>Interpretation:</b> The correlation of 0.67 shows that there is positive significant relationship between tip and total_bill indicating the significant increase in total bill as the increase in tip occurs.

# Main Approaches(Methods) for calculating correlation 

1. Pearson's correlation coefficient
2. Spearman's rank correlation coefficient

# Pearson's correlation coefficient

Pearson's correlation coefficient is a measure of the linear correlation between two variables X and Y. It has a value between +1 and -1 where 1 is total positive linear correlation, 0 is no linear correlation, and -1 is total negative correlation. It is applied assuming that the dataset is normally distributed. It is a parametric test.

# Example of Pearson's correlation coefficient from formula without using library

In [3]:
# pearson's correlation coefficient
import numpy as np
import pandas as pd

def pearson(x,y):
    x = np.array(x)
    y = np.array(y)
    if len(x) != len(y):
        raise ValueError("The length of x and y must be the same.")
    x_mean = np.mean(x) 
    y_mean = np.mean(y) 
    x_std = np.std(x)
    y_std = np.std(y)
    n = len(x) 
    return sum((x - x_mean) * (y - y_mean)) / (n * x_std * y_std)

In [5]:
# example dataset 
x = np.array([1,2,3,4,5]) 
y = np.array([2,3,7,8,6]) 
# calculating pearson's correlation coefficient 
pearson_coefficient = pearson(x, y) 
print(f"Pearson's correlation coefficient: {pearson_coefficient}") 

Pearson's correlation coefficient: 0.794101388315984


<b>Interpreations:</b><br>1. Pearson's correlation coefficient is 0.79 which shows the positive correlation between x and y variable.<br>2.The dependent and independent variables are not well defined.